.eleventy.js

```diff
@@ -1,7 +1,19 @@
 const buildTOC = require('./src/BuildTOC')
 
-module.exports = function(eleventyConfig, pluginNamespace) {
-  eleventyConfig.namespace(pluginNamespace, () => {
-    eleventyConfig.addFilter('toc', (content, opts) => buildTOC(content, opts))
-  })
-}
+module.exports = (function() {
+  let _opts = {}
+  function TOC(eleventyConfig, pluginNamespace) {
+    eleventyConfig.namespace(pluginNamespace, () => {
+      eleventyConfig.addFilter('toc', (content, opts) => {
+        return buildTOC(content, {..._opts, ...opts})
+      });
+    });
+  }
+
+  TOC.opts = function(opts) {
```

Comment:
I'd prefer the more standard
`eleventyConfig.addPlugin(pluginTOC({}))`

rather than this approach
`eleventyConfig.addPlugin(pluginToc.opts({}))`

Can we redo this file to:
```js
const buildTOC = require('./src/BuildTOC')
const parseOptions = require('./src/ParseOptions')

module.exports = globalOpts => (eleventyConfig, pluginNamespace) => {
  eleventyConfig.namespace(pluginNamespace, () => {
    eleventyConfig.addFilter('toc', (content, localOpts) => {
      return buildTOC(
        content,
        parseOptions(localOpts, globalOpts)
      )
    });
  });
}
```

That will provide a plugin config that's more common in Eleventy (at least in my experience - please let me know if you've seen a different pattern more common).

Additionally, using `parseOptions()` here will allow Nunjucks users a more fault-tolerant template config. Both of these will work:
```njk
<!-- raw JSON -->
{{ content | toc({"wrapper": "div"}) | safe }}
<!-- quoted JSON -->
{{ content | toc('{"wrapper": "div"}') | safe }}
```

Reply:
Couldn't `eleventyConfig.addPlugin(pluginTOC({}))` break for users who have it configured as `eleventyConfig.addPlugin(pluginTOC)`. I'm not sure if that could or be should be classified as a breaking change, would you want that?

Though the use of `parseOptions` in the `.eleventy.js` file make sense to me.

Reply:
@jdsteinbach ok, I think I have a clean solution to what you mentioned about not having the default options as a function of the TOC plugin. Apparently the [second argument of `addPlugin` is for options](https://www.11ty.dev/docs/plugins/#plugin-configuration-options), so both

```js
module.exports = function (eleventyConfig) {
  eleventyConfig.addPlugin(pluginTOC)
}
```

and 

```js
module.exports = function (eleventyConfig) {
  eleventyConfig.addPlugin(pluginTOC, {
    tags: ['h2', 'h3'],
    wrapper: 'div'
  })
}
```

should work, and are the 11ty way of doing plugins.